### PR TITLE
Move drag handle outside the cell

### PR DESCRIPTION
**Pull request by amanmahajan7**
Pros:
- Cleans up row and cell props

Cons
- Needs special handling for col spans

### CODE REVIEW

File: src/DragHandle.tsx [hunk 1]
```diff
@@ -105,7 +121,15 @@ export default function DragHandle<R, SR>({
 
   return (
     <div
-      className={cellDragHandleClassname}
+      style={{
+        gridColumnStart: column.idx + 1,
+        gridRowStart,
+        insetInlineStart:
+          column.frozen && typeof column.width === 'number'
```
Comment (amanmahajan7) [inline src/DragHandle.tsx:128]: Not sure if this can cause any issues

File: src/DragHandle.tsx [hunk 2]
```diff
@@ -105,7 +121,15 @@ export default function DragHandle<R, SR>({
 
   return (
     <div
-      className={cellDragHandleClassname}
+      style={{
+        gridColumnStart: column.idx + 1,
```
Comment (amanmahajan7) [inline src/DragHandle.tsx:125]: Need to handle `colSpan`. I will check in the next PR

Comment (amanmahajan7) [reply src/DragHandle.tsx]: Fix in this PR

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -799,13 +799,18 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getDragHandle(rowIdx: number) {
-    if (selectedPosition.rowIdx !== rowIdx || selectedPosition.mode === 'EDIT' || onFill == null) {
+  function renderDragHandle() {
+    if (
```
Comment (nstepien) [inline src/DataGrid.tsx:803]: Do we need to check whether the selected cell is not in a summary row?

File: src/DragHandle.tsx [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
-    position: absolute;
-    inset-inline-end: 0;
-    inset-block-end: 0;
-    inline-size: 8px;
-    block-size: 8px;
+    inline-size: var(--rdg-drag-handle-size);
+    block-size: var(--rdg-drag-handle-size);
     background-color: var(--rdg-selection-color);
+    align-self: end;
+    justify-self: end;
```
Comment (nstepien) [inline src/DragHandle.tsx:17]:
```suggestion
    place-self: end;
```

File: src/DataGrid.tsx (hunk 3 above)
Comment (amanmahajan7) [reply src/DataGrid.tsx:803]: `isCellWithinViewportBounds` would return false for header and summary rows. There is another function `isCellWithinSelectionBounds` that can be used to check header and summary rows as well

**Review by nstepien — APPROVED** [review]

**Pull request merged**